precompile/contracts/warp/contract.go

```diff
@@ -0,0 +1,268 @@
+// Code generated
+// This file is a generated precompile contract config with stubbed abstract functions.
+// The file is generated by a template. Please inspect every code and comment in this file before use.
+
+package warp
+
+import (
+	"fmt"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/vms/platformvm/warp"
+	"github.com/ava-labs/subnet-evm/params"
+	"github.com/ava-labs/subnet-evm/precompile/contract"
+	"github.com/ava-labs/subnet-evm/vmerrs"
+	warpMessages "github.com/ava-labs/subnet-evm/warp/messages"
+
+	_ "embed"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	GetBlockchainIDGasCost uint64 = 2 // Based on GasQuickStep used in existing EVM instructions
+	// Cost of fixed log size and writing log to database (Note: this calculation includes buffer room because a warp message
+	// is not stored in the trie and not stored permanently)
+	SendWarpMessageGasCost uint64 = params.LogGas + 4*params.LogTopicGas + contract.WriteGasCostPerSlot
+	// TODO: Gas cost per byte in the payload/log data of the produced warp message
+	SendWarpMessageGasCostPerByte uint64 = 1
+
+	GetVerifiedWarpMessageGasCost uint64 = 1 // TODO: base this off of the number of signers
+	GasCostPerWarpSigner          uint64 = 1 // TODO
+	GasCostPerWarpMessageBytes    uint64 = 1 // TODO
+)
```

Comment:
TODO: need to benchmark the warp precompile predicate and execution in order to calculate accurate gas costs.